File: SpringTILCode/src/test/java/com/example/springtilcode/user/AuthServiceTest.java
package com.example.springtilcode.user;

import com.example.springtilcode.domain.user.controller.dto.TokenResponse;
import com.example.springtilcode.domain.user.service.AuthService;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

@SpringBootTest
public class AuthServiceTest {

    @Autowired
    private AuthService authService;

    @Test
    @DisplayName("signup")
    public void signup(){
        authService.signup("test@gmail.com", "1234");
    }

    @Test
    @DisplayName("login")
    public void login(){
        authService.login("test@gmail.com", "1234");
    }
}

File: kotlin/src/test/kotlin/com/example/demo/user/AuthServiceTest.kt
package com.example.demo.user

import com.example.demo.domain.user.repository.UserRepository
import com.example.demo.domain.user.service.AuthService
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest

@SpringBootTest
class AuthServiceTest (
    @Autowired
    val authService: AuthService,
    @Autowired
    val userRepository: UserRepository
){
    @Test
    @DisplayName("1. 회원가입")
    fun signup(){
        val email = "whee@gmail.com"
        val password = "1234"

        val userResponse = authService.signup(email, password)
        val findUser = userRepository.findByEmail(email)

        if (findUser != null) {
            Assertions.assertEquals(findUser.email, userResponse.email)
        }
    }

    @Test
    @DisplayName("2. 로그인")
    fun login(){
        val email = "whee@gmail.com"
        val password = "1234"

        val tokenResponse = authService.login(email, password)
        println(tokenResponse)
    }
}